src/main/java/org/minions/devfund/abel/battleship/Ship.java

```diff
@@ -0,0 +1,225 @@
+package org.minions.devfund.abel.battleship;
+
+/**
+ * Class to handle Ship.
+ */
+public abstract class Ship {
+
+    private int bowRow;
+    private int bowColumn;
+    private int length;
+    private boolean horizontal;
+    private boolean[] hit;
+    private String shipType = this.getClass().getSimpleName().toLowerCase();
+
+    /**
+     * Get Bow Row.
+     *
+     * @return int.
+     */
+    public int getBowRow() {
+        return bowRow;
+    }
+
+    /**
+     * Set Bow Row.
+     *
+     * @param bowRow int.
+     */
+    public void setBowRow(int bowRow) {
+        this.bowRow = bowRow;
+    }
+
+    /**
+     * Get Bow Column.
+     *
+     * @return int.
+     */
+    public int getBowColumn() {
+        return bowColumn;
+    }
+
+    /**
+     * Set Bow Column.
+     *
+     * @param bowColumn int.
+     */
+    public void setBowColumn(int bowColumn) {
+        this.bowColumn = bowColumn;
+    }
+
+    /**
+     * Get Ship Length.
+     *
+     * @return int.
+     */
+    public int getLength() {
+        return length;
+    }
+
+    /**
+     * Set Ship Length.
+     *
+     * @param length int.
+     */
+    public void setLength(int length) {
+        this.length = length;
+    }
+
+    /**
+     * If the Ship is Horizontal or not.
+     *
+     * @return boolean.
+     */
+    public boolean isHorizontal() {
+        return horizontal;
+    }
+
+    /**
+     * Set Ship Horizontal.
+     *
+     * @param horizontal boolean.
+     */
+    public void setHorizontal(boolean horizontal) {
+        this.horizontal = horizontal;
+    }
+
+    /**
+     * Get Ship Hit status.
+     *
+     * @return boolean[].
+     */
+    public boolean[] getHit() {
+        return hit.clone();
+    }
+
+    /**
+     * Set Ship Hit Status.
+     *
+     * @param hit boolean[].
+     */
+    public void setHit(final boolean[] hit) {
+        this.hit = hit.clone();
+    }
+
+    /**
+     * Get Ship Type.
+     *
+     * @return String.
+     */
+    public String getShipType() {
+        return shipType;
+    }
+
+    /**
+     * If its ok to place the Ship at provided position.
+     *
+     * @param row        int.
+     * @param column     int.
+     * @param horizontal boolean.
+     * @param ocean      Ocean.
+     * @return boolean.
+     */
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)